diff --git a/vesc_ackermann/src/vesc_to_odom.cpp b/vesc_ackermann/src/vesc_to_odom.cpp
--- a/vesc_ackermann/src/vesc_to_odom.cpp
+++ b/vesc_ackermann/src/vesc_to_odom.cpp
@@ -55,8 +55,9 @@ VescToOdom::VescToOdom(const rclcpp::NodeOptions & options)
 {
   // get ROS parameters
-  declare_parameter("odom_frame", odom_frame_);
-  declare_parameter("base_frame", base_frame_);
-  declare_parameter("use_servo_cmd_to_calc_angular_velocity", use_servo_cmd_);
+  odom_frame_ = declare_parameter<std::string>("odom_frame", odom_frame_);
+  base_frame_ = declare_parameter<std::string>("base_frame", base_frame_);
+  use_servo_cmd_ = declare_parameter<bool>(
+    "use_servo_cmd_to_calc_angular_velocity", use_servo_cmd_);
 
   speed_to_erpm_gain_ = declare_parameter<double>("speed_to_erpm_gain");
   speed_to_erpm_offset_ = declare_parameter<double>("speed_to_erpm_offset");
@@ -72,7 +73,7 @@ VescToOdom::VescToOdom(const rclcpp::NodeOptions & options)
     wheelbase_ = declare_parameter<double>("wheelbase");
   }
 
-  declare_parameter("publish_tf", publish_tf_);
+  publish_tf_ = declare_parameter<bool>("publish_tf", publish_tf_);
 
   // create odom publisher
   odom_pub_ = create_publisher<Odometry>("odom", 10);
